ocs_ci/ocs/resources/bucket_policy.py

```
@@ -183,14 +185,23 @@ def gen_bucket_policy(
         resource_property (str):  Element specifies the object(s) that the statement covers
+        action_prefix (str): Service prefix for actions (default: "s3"). Use "s3vectors" for
+            vector bucket policies.
+        resource_arn_service (str): Service name in ARN (default: "s3"). Use "s3vectors" for
+            vector bucket policies. Resources that already start with "arn:" are used as-is.
 
     Returns:
         dict: Bucket policy in json format
     """
     principals = user_list
-    actions = list(map(lambda action: "s3:%s" % action, actions_list))
+    actions = list(map(lambda action: f"{action_prefix}:{action}", actions_list))
```

Ruff (0.15.18)

[warning] Unnecessary `map()` usage (rewrite using a list comprehension)

Replace `map()` with a list comprehension

(C417)